scene/3d/skeleton_3d.cpp

```diff
@@ -1116,6 +1116,12 @@ void Skeleton3D::_force_update_bone_children_transforms(int p_bone_idx) const {
 
 	// Loop through nested set.
 	for (int offset = 0; offset < bone_size; offset++) {
+		if (rest_dirty) {
+			int current_bone_idx = nested_set_offset_to_bone_index[offset];
+			Bone &b = bonesptr[current_bone_idx];
+			b.global_rest = b.parent >= 0 ? bonesptr[b.parent].global_rest * b.rest : b.rest; // Rest needs update apert from pose.
+		}
+
```

Comment:
```suggestion
		int current_bone_idx = nested_set_offset_to_bone_index[offset];
		Bone &b = bonesptr[current_bone_idx];
		bool bone_enabled = b.enabled && !show_rest_only;
		if (rest_dirty) {
			b.global_rest = b.parent >= 0 ? bonesptr[b.parent].global_rest * b.rest : b.rest; // Rest needs update apert from pose.
		}

```

`current_bone_idx` and `&b` are defined again immediately below.

I think we wanted to skip access to the arrays for improved performance if there is no pose change, so I put them inside the if block to account for the less frequency of rest dirty, but the code would look cleaner if we put them outside. Which would be better?

Reply:
> so I put them inside the if block to account for the frequency of rest dirty (since it should be few)

I wonder if we can see the performance on a graph.